FiftyOne.Pipeline.Elements/FiftyOne.Pipeline.JavaScriptBuilderElementTests/CookiesTests.cs

```diff
@@ -0,0 +1,143 @@
+                        true)
+                };
+
+            protected override void ProcessInternal(IFlowData data)
+            {
+                var result = new CookieData(_loggerFactory.CreateLogger<CookieData>(), data.Pipeline);
+                result["javascript"] = "document.cookie =  \"some cookie value\"";
+                data.GetOrAdd(ElementDataKey, p => result);
+            }
+
+            protected override void ManagedResourcesCleanup()
+            {
+            }
+
+            protected override void UnmanagedResourcesCleanup()
+            {
+            }
+        }
+
+        /// <summary>
+        /// Test various configurations for enabling cookies to verify
+        /// that cookies are/aren't written for each configuration.
+        /// 
+        /// The source JavaScript contains code to set a cookie. The JSBuilder
+        /// element should replace this if the config says that cookies are not
+        /// enabled.
+        /// </summary>
+        /// <param name="enableInConfig">
+        /// True if cookies are enabled in the element configuration.
+        /// </param>
+        /// <param name="enableInEvidence">
+        /// True if cookies are enabled in the evidence.
+        /// </param>
+        /// <param name="expectCookie">
+        /// True if the test should expect cookies to be enabled for this configuration.
+        /// </param>
+        [DataRow(false, false, false)]
+        [DataRow(true, false, false)]
+        [DataRow(false, true, true)]
+        [DataRow(true, true, true)]
+        [DataTestMethod]
+        public void TestJavaScriptCookies(bool enableInConfig, bool enableInEvidence, bool expectCookie)
+        {
+            // Arrange
+            var cookieElement = new CookieElement(_loggerFactory);
+            var sequenceElement = new SequenceElementBuilder(_loggerFactory)
+                .Build();
+            var jsonElement = new JsonBuilderElementBuilder(_loggerFactory)
+                .Build();
+            var jsElement = new JavaScriptBuilderElementBuilder(_loggerFactory)
+                .SetEnableCookies(enableInConfig)
+                .Build();
+            var pipeline = new PipelineBuilder(_loggerFactory)
+                .AddFlowElement(cookieElement)
+                .AddFlowElement(sequenceElement)
+                .AddFlowElement(jsonElement)
+                .AddFlowElement(jsElement)
+                .Build();
+
+            // Act
+            string javaScript = null;
+            using (var flowData = pipeline.CreateFlowData())
+            {
+                flowData.AddEvidence(
+                    JavaScriptBuilder.Constants.EVIDENCE_ENABLE_COOKIES,
+                    enableInEvidence.ToString());
+                flowData.Process();
+                javaScript = flowData.Get<IJavaScriptBuilderElementData>().JavaScript;
+            }
+
+            // Assert
+            var cookieRegex = new Regex("document\\.cookie *= *");
```

Comment:
@ben51degrees this is a bit too harsh a test :) because in the served javascript the snippet received as part of json will be present unchanged, it will be altered during the execution of the javascript - it will replace document.cookie with sessionStorage.

We need to test that depending enableCookies = true or false a different variant of javascript is produced from the template...  Once we merge the `javascript-templates` [PR](https://github.com/51Degrees/pipeline-dotnet/pull/95) @drasmart suggests a test to check the number of occurrences of `document.cookie` for the case of enableCookies=true (should be 2) and false (should be 1): 

![image](https://github.com/51Degrees/pipeline-dotnet/assets/341967/cd533d5c-12e0-43ce-9eb5-1dfe8ad29f1c)

Reply:
@drasmart please apply this change once other template-related PRs are merged as PR to Ben's branch.